src/app/loadout-builder/mod-utils.ts

```diff
@@ -26,46 +24,47 @@ export const doEnergiesMatch = (mod: LockedArmor2Mod, item: DimItem) =>
  * Assignes the general mods to armour pieces in assignments, including the energy specific ones
  * i.e. Void Resist ect
  *
- * assignments is mutated in this function as it tracks assigned mods for a particular armour set
+ * assignments is mutated in this function as it tracks assigned mods for a particular armour set.
+ *
+ * This function need to be kept inline with processWorker/processUtils#canTakeAllGeneralMods.
  */
 function assignGeneralMods(
   setToMatch: readonly DimItem[],
   generalMods: LockedArmor2Mod[],
   assignments: Record<number, LockedArmor2Mod[]>
 ): void {
-  const armour2Items = setToMatch.filter((item) => item.isDestiny2() && item.energy);
-  if (generalMods && armour2Items.length < generalMods.length) {
-    return;
-  }
-
-  const generalModsByEnergyType = _.groupBy(
-    generalMods,
-    (mod) => mod.mod.plug.energyCost.energyType
-  );
+  const sortedItems = [...setToMatch].sort((a, b) => {
```

Comment:
We should be consistent about `[...list]` vs `Array.from(list)`

Reply:
I am going to assume Array.from is preferred.